crates/bevy_ecs/src/world/unsafe_world_cell.rs

```diff
@@ -650,9 +650,9 @@ impl Debug for UnsafeWorldCell<'_> {
 /// A interior-mutable reference to a particular [`Entity`] and all of its components
 #[derive(Copy, Clone)]
 pub struct UnsafeEntityCell<'w> {
-    world: UnsafeWorldCell<'w>,
-    entity: Entity,
-    location: EntityLocation,
+    pub(crate) world: UnsafeWorldCell<'w>,
+    pub(crate) entity: Entity,
```

Comment:
Is it really necessary to loosen the privacy here?  

It looks like this is only needed so you can get an `&Entity` instead of an `Entity` to `impl Borrow<Entity>` for a few types.  Since we're no longer using `Borrow<Entity>` internally, could we just leave out those impls?  

Alternately, could we add a `pub (crate)` method to get `&Entity` so that it's clear nothing outside of this module modifies these fields?

Reply:
right, there is a much simpler solution: I hadn't thought of just adding an `entity` method on `UnsafeEntityCell` itself for some reason. Should `UnsafeEntityCell` itself implement `EntityBorrow`?

Reply:
Ah, I remembered why I didn't do this initially. Unless you return `&Entity` from a field, you will get a temporary lifetime error. So unless we increase the visibility here, or move the type to `entity_ref.rs`, we cannot impl `Borrow<Entity>` for the `EntityRef` types.
The reason why I added this impl is for the original purpose of `Borrow`: to allow its use in `Map` and `Set` types.
But I can omit it instead then.

Reply:
I think it works if the method returns a reference to the original value: 

```rust
impl UnsafeEntityCell<'_> {
    pub(crate) fn entity_ref(&self) -> &Entity {
        &self.entity
    }
}

impl Borrow<Entity> for EntityMut<'_> {
    fn borrow(&self) -> &Entity {
        self.0.entity_ref()
    }
}
```

But it's a moot point now that you removed them.

Reply:
ahhh, you're right! Had I taken my train of thought just a bit further I could've realized, we can just impl `Borrow<Entity>` and `EntityBorrow` for `UnsafeEntityCell`...
I'll add them back.